src/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs

```diff
@@ -85,6 +85,7 @@ namespace System.Reflection.Metadata
     public readonly partial struct ArrayShape
     {
         private readonly object _dummy;
+        private readonly int _dummyPrimitive;
```

Comment:
cc @nguerrera, @tmat to review the Reflection.Metadata changes. 

Spot checked and seems to make sense:
https://github.com/dotnet/corefx/blob/16f3eef83d6df4c9dbc0e2992dafbc9553db363d/src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/AssemblyFile.cs#L11-L14
https://github.com/dotnet/corefx/blob/16f3eef83d6df4c9dbc0e2992dafbc9553db363d/src/System.Reflection.Metadata/src/System/Reflection/Metadata/Signatures/CustomAttributeNamedArgument.cs#L7-L12
https://github.com/dotnet/corefx/blob/16f3eef83d6df4c9dbc0e2992dafbc9553db363d/src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/ExportedType.cs#L9-L14